src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -980,15 +980,19 @@ export class FileDragAndDrop implements ITreeDragAndDrop<ExplorerItem> {
 
 		try {
 
-			// Desktop DND (Import file)
+			// External file DND (Import/Upload file)
 			if (data instanceof NativeDragAndDropData) {
-				if (isWeb) {
+				// Native OS file DND into Web
+				if (originalEvent.dataTransfer?.types.indexOf('Files') !== -1 && isWeb) {
```

Comment:
I know we use `originalEvent.dataTransfer?.types.indexOf('Files')` 2 times in this file now but it seems like we could extract this into a method for reuse? And is there a spec that explains what this does?

An alternative would be to add a static method to `BrowserFileUpload` to ask wether it would apply for the given event.

Reply:
https://html.spec.whatwg.org/multipage/dnd.html#dom-datatransfer-types-dev